### PR TITLE
GAMs new api

### PR DESCRIPTION
**Pull request by arokem**

### CODE REVIEW

**Comment by arokem** [comment]
This PR is still WIP, and will be a place where we overhaul the GAM API. For now, it's just refactoring existing code in preparation for that.

**Comment by arokem** [comment]
This has already gotten pretty big, and seems to be passing testing now, so I suggest that we merge it before we start work on another PR for API updates.

File: DESCRIPTION [hunk 1]
```diff
@@ -1,25 +1,50 @@
...
            role = c("aut", "cre"),
-           comment = c(ORCID = "0000-0001-9276-9084"))
-Description: A toolbox for analyzing diffusion MRI tract profiles using
-    generalized additive models.
+           comment = c(ORCID = "0000-0001-9276-9084")),
+    person(given = "Kelly",
+           family = "Chang",
+           email = "XXX",
+           role = c("aut"),
+           comment = c(ORCID = "XXX")),
```
Comment (arokem) [inline DESCRIPTION:14]: @kellychang4 : what email would you like to use here, and what's your ORCID?

File: DESCRIPTION [hunk 2]
```diff
@@ -1,25 +1,50 @@
...
+    person(given = "Teresa",
+           family = "Gomez",
+           email = "XXX",
+           role = c("aut"),
+           comment = c(ORCID = "XXX")),
```
Comment (arokem) [inline DESCRIPTION:19]: @teresamg : what email would you like to use here, and what's your ORCID?

File: DESCRIPTION [hunk 3]
```diff
@@ -1,25 +1,50 @@
...
+    person(given = "McKenzie",
+           family = "Hagen",
+           email = "XXX",
+           role = c("aut"),
+           comment = c(ORCID = "XXX")),
```
Comment (arokem) [inline DESCRIPTION:24]: @mckenziephagen : what email would you like to use here, and what's your ORCID?

Comment (mckenziephagen) [reply DESCRIPTION:24]: ORCID: 0000-0002-7454-8189
email: mphagen@uw.edu

**Pull request merged**